File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int l = 0, r = 0;
        int n = nums.size();
        int ans = 0, res = 0;

        unordered_map<int, int> map;

        for (; r < n; r++) {
            map[nums[r]]++;
            ans += nums[r];

            while (map[nums[r]] >= 2)
                map[nums[l]]--, ans -= nums[l++];
                
            res = fmax(res, ans);
        }

        return res;
    }
};

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int n = nums.length;
        int l = 0, r = 0;
        int temp = 0, max = 0;

        var map = new int[10001];
        for (; r < n; r++) {
            map[nums[r]]++;
            temp += nums[r];
            
            while (map[nums[r]] >= 2) {
                temp -= nums[l];
                map[nums[l++]]--;
            }

            max = Math.max(max, temp);
        }
        return max;
    }
}